gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -462,9 +459,9 @@ private static Set<SourceEntity> getSourceEntities(String response) {
   }
 
   protected SalesforceConnector getConnector(State state) {
-    if (this.salesforceConnector == null) {
-      this.salesforceConnector = new SalesforceConnector(state);
+    if (salesforceConnector == null) {
+      salesforceConnector = new SalesforceConnector(state);
```

Comment:
this is not thread-safe... should it be `synchronized`?  (if there's any need to explicitly close the connector, there's a potential resource leak.)